### DISCUSSION TITLE
Thought on Responsive Framework for Main Template/UX

### DISCUSSION THREAD

**Discussion by vicos59** [geodesicsolutions-community/geocore-community#63] [Ideas]
A thread to discuss making the main template set adhere to modern standards and perhaps use an open source Framework like Bootstrap. How gigantic of a task would this be? My only experience is with Bootstrap. I converted quite a few legacy systems and it was quite the task.

**Comment by jonyo** [comment]
It already uses bootstrap, and is already responsive.  That was actually done back in 2013 [you can see here in version 7.3](https://geodesicsolutions.org/changelog/7.3/changelog.html#bug_254).

See:
<img width="321" alt="Screen Shot 2022-02-09 at 08 19 33" src="https://user-images.githubusercontent.com/2767294/153219739-051d6c8c-8d60-40a5-8332-19923664e034.png">

**Reply by jonyo** [reply]
I mean I would not be against updating it, the one in the software now uses a much earlier version of bootstrap now.  Just saying it does already use it and is already responsive, unless you were referring to a specific thing?  Or maybe you were talking about the admin panel?

I also would be for updating the JS used behind the scenes to use something like vuejs to replace jquery...

**Comment by vicos59** [comment]
I've only ever used Fusion for v7.5.1 and I know it is Bootstrap based, but for some reason I thought I read that it was only using a small subset of Bootstrap. In any case many years I spent a very brief amount of time looking at updating it to use our Bootstrap config and it would not have been a trivial update.

In any case Bootstrap v5.1 is out. However, I would not hold up the first production release of CoreCE to implement this. Have to draw the line somewhere and the current templates still work.

**Reply by jonyo** [reply]
> However, I would not hold up the first production release of CoreCE to implement this. 

Agreed, anything that requires a large rewrite I would hold off on.  Especially in this case since it would possibly require a lot of work for people with custom designs.  We want the update to v20 to be as easy as possible since it is the first "official" CE release.
